frontend/src/container/AdvancedSearch.jsx

```diff
@@ -175,6 +187,167 @@ export class AdvancedSearch extends React.Component {
     }
   }
 
+  handleChangeSearchFieldList = (field) => {
+    const { props, state } = this
+    let currentSearch
+    if (state.searchType === ADVANCED_SEARCH_TYPE.CONTENT) {
+      currentSearch = props.contentSearch
+    }
+    /*
+      if (state.searchType === ADVANCED_SEARCH_TYPE.USER) {
+        currentSearch= props.userSearch
+      }
+
+      if (state.searchType === ADVANCED_SEARCH_TYPE.SPACE) {
+        currentSearch = props.spaceSearch
+      }
+    */
+    let newAppliedSearchFieldList = []
+    const oldAppliedSearchFieldList = currentSearch.appliedFilterList.searchFieldList
+
+    if (oldAppliedSearchFieldList) {
+      if (oldAppliedSearchFieldList.find(searchField => searchField === field.slug)) {
+        newAppliedSearchFieldList = oldAppliedSearchFieldList.filter(searchField => searchField !== field.slug)
+      } else newAppliedSearchFieldList = [...oldAppliedSearchFieldList, field.slug]
+    } else newAppliedSearchFieldList = [field.slug]
+
+    props.dispatch(setAppliedFilter(ADVANCED_SEARCH_FILTER.SEARCH_FIELD, newAppliedSearchFieldList, state.searchType))
+
+    this.getSearchResult(
+      { ...currentSearch, searchType: state.searchType },
+      currentSearch.resultList.length,
+      newAppliedSearchFieldList,
+      currentSearch.appliedFilterList.createdRange,
+      currentSearch.appliedFilterList.modifiedRange,
+      currentSearch.appliedFilterList.searchFacets
+    )
+  }
+
+  handleChangeCreatedRange = (dateObject) => {
+    const { props, state } = this
+    let currentSearch
+
+    if (state.searchType === ADVANCED_SEARCH_TYPE.CONTENT) {
+      currentSearch = props.contentSearch
+    }
+    /*
+      if (state.searchType === ADVANCED_SEARCH_TYPE.USER) {
+        currentSearch= props.userSearch
+      }
+
+      if (state.searchType === ADVANCED_SEARCH_TYPE.SPACE) {
+        currentSearch = props.spaceSearch
+      }
+    */
+
+    let newAppliedCreatedRange = {}
+    const oldAppliedCreatedRange = currentSearch.appliedFilterList.createdRange
+
+    if (oldAppliedCreatedRange) {
+      if (Object.keys(dateObject)[0] in oldAppliedCreatedRange) {
```

Comment:
What is `Object.keys(dateObject)[0]`? The modified property during the last user interaction, right? (maybe put it in a const?)

 Are you sure you will get the right key? (there is only one property in `dateObject`?)

Maybe avoid `in`:
```suggestion
      if (Object.keys(oldAppliedCreatedRange).includes(Object.keys(dateObject)[0])) {
```

Reply:
Since filters are applied one at a time, there will only be one property on the object each time. I will make the change and put it in a const for clarity.